src/main/webapp/app/complaints/complaints-for-students/complaints-student-view.component.ts

```diff
@@ -143,8 +144,12 @@ export class ComplaintsStudentViewComponent implements OnInit {
         if (!actionThresholdInDays) {
             return false;
         }
-        if (!this.exercise.assessmentDueDate || dayjs().isAfter(dayjs(this.exercise.assessmentDueDate))) {
-            return dayjs().isSameOrBefore(dayjs(completionDate).add(actionThresholdInDays, 'day'));
+        const isWithinThreshold = dayjs().isSameOrBefore(dayjs(completionDate).add(actionThresholdInDays, 'day'));
+        if (!this.exercise.assessmentDueDate) {
+            return isWithinThreshold;
+        }
+        if (dayjs().isAfter(dayjs(this.exercise.assessmentDueDate))) {
+            return isWithinThreshold || dayjs().isBefore(dayjs(this.exercise.assessmentDueDate).add(actionThresholdInDays, 'day'));
```

Comment:
You used isSameOrBefore in 147 but only used isBefore in 152. Is that on purpose?

Reply:
No, good catch